core/controllers/base.py

```diff
@@ -571,6 +572,21 @@ def _render_exception_json_or_html(self, return_type, values):
                     'error-iframed.mainpage.html', iframe_restriction=None)
             elif values['status_code'] == 503:
                 self.render_template('maintenance-page.mainpage.html')
+            elif values['status_code'] == 404:
+                # Routes for pages are handled at the frontend using angular
+                # router. As they are not registered at the backend,
+                # the backend tries to return error page with 404 status code.
+                # This scenario is handled here, so that pages are served with
+                # appropriate status code.
+
+                route_is_handled_using_angular_router = any(
+                    self.request.path == '/%s' %
+                    route for route in constants.FRONTEND_ROUTES.values())
```

Comment:
Perhaps FRONTEND_ROUTES --> ROUTES_HANDLED_BY_FRONTEND might be clearer.

Reply:
I think ROUTES_REGISTERED_WITH_FRONTEND will be better.

Reply:
changed to `ROUTES_REGISTERED_WITH_FRONTEND`